src/gluonts/distribution/beta.py

```diff
@@ -69,14 +69,20 @@ def log_prob(self, x: Tensor) -> Tensor:
         F = self.F
         alpha, beta = self.alpha, self.beta
 
-        return (
-            (alpha - 1) * F.log(x)
-            + (beta - 1) * F.log(1 - x)
+        # add machine epsilon to avoid infinities due to 0 padding which is masked,
+        # however, causes issues in backpropagation
+        epsilon = np.finfo(self._dtype).eps
+
+        ll = (
+            (alpha - 1) * F.log(x + epsilon)
+            + (beta - 1) * F.log(1 - x + epsilon)
```

Comment:
I don’t get it: why is this needed?

If a value outside of the domain ends up in x due to padding, then masking should take care of the inf or nan that comes out of this. Otherwise I don’t think we should try to recover from 0s or 1s in the data (one should validate the data beforehand)